src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/Airship.cs

```diff
@@ -93,6 +96,48 @@ public string NamedUser
             }
         }
 
+        private ChannelHandler onChannelCreation;
+        public event ChannelHandler OnChannelCreation
+        {
+            add
+            {
+                Console.WriteLine("onChannelCreationAdd");
+                onChannelCreation += value;
+                UAirship.Shared().Channel.AddChannelListener(this);
```

Comment:
Pretty sure you cant follow this pattern since its more than one listener. You should add the listener once in the constructor or an init method otherwise you will send duplicate events for every added listener. Try adding two listeners to see what I mean.

Reply:
If I understand I have to add listener method (AddChannelListener) in an existing method like constructor or init method to don't have duplicate calls ?

Reply:
Yeah, so in the constructor add the listener once, never remove it.